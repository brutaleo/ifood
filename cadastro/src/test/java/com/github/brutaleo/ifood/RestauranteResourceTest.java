package com.github.brutaleo.ifood;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.given;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
// previne as checagens do postgres vs as definições no arquivo *.yml
@QuarkusTest
@QuarkusTestResource(CadastroTestLifeCycleManager.class)
public class RestauranteResourceTest {

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    public void RestaurantesEndpointTest() {
        String resultado = given()
                .when().get("/restaurantes")
                .then()
                .statusCode(200)
                .extract()
                .asString();
        Approvals.verifyJson(resultado);
    }

}
