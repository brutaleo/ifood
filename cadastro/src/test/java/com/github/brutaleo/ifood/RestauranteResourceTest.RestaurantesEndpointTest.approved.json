[
  {
    "id": 123,
    "nome": "Nova População de restaurantes 1",
    "proprietario": "Id no keyclock 1"
  },
  {
    "id": 456,
    "nome": "Nova População de restaurantes 2",
    "proprietario": "Id no keyclock 2"
  }
]